check_license/check_headers.sh

```diff
@@ -0,0 +1,191 @@
+#!/usr/bin/env bash
+# Copyright (C) 2016-2025 Intel Corporation
+# Under the Apache License v2.0 with LLVM Exceptions. See LICENSE.TXT.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+# check-headers.sh - check copyright and license in source files
+
+SELF=$0
+
+function usage() {
+	echo "Usage: $SELF <source_root_path> <license_tag> [-h|-v|-a|-d]"
+	echo "   -h, --help          this help message"
+	echo "   -v, --verbose       verbose mode"
+	echo "   -a, --all           check all files (only modified files are checked by default)"
+	echo "   -d, --update_dates  change Copyright dates in all analyzed files (rather not use with -a)"
+}
+
+if [ "$#" -lt 2 ]; then
+	usage >&2
+	exit 2
+fi
+
+SOURCE_ROOT=$1
+shift
+LICENSE=$1
+shift
+
+PATTERN=`mktemp`
+TMP=`mktemp`
+TMP2=`mktemp`
+TEMPFILE=`mktemp`
+rm -f $PATTERN $TMP $TMP2
+
+if [ "$1" == "-h" -o "$1" == "--help" ]; then
+	usage
+	exit 0
+fi
+
+export GIT="git -C ${SOURCE_ROOT}"
+$GIT rev-parse || exit 1
+
+if [ -f $SOURCE_ROOT/.git/shallow ]; then
+	SHALLOW_CLONE=1
+	echo
+	echo "Warning: This is a shallow clone. Checking dates in copyright headers"
+	echo "         will be skipped in case of files that have no history."
+	echo
+else
+	SHALLOW_CLONE=0
+fi
+
+VERBOSE=0
+CHECK_ALL=0
+UPDATE_DATES=0
+while [ "$1" != "" ]; do
+	case $1 in
+	-v|--verbose)
+		VERBOSE=1
+		;;
+	-a|--all)
+		CHECK_ALL=1
+		;;
+	-d|--update_dates)
+		UPDATE_DATES=1
+		;;
+	esac
+	shift
+done
+
+if [ $CHECK_ALL -eq 0 ]; then
+	CURRENT_COMMIT=$($GIT log --pretty=%H -1)
+	MERGE_BASE=$($GIT merge-base HEAD origin/master 2>/dev/null)
+	[ -z $MERGE_BASE ] && \
+		MERGE_BASE=$($GIT log --pretty="%cN:%H" | grep GitHub | head -n1 | cut -d: -f2)
```

Comment:
Here is a broken pipe. Why? Because `grep` finishes work and the data doesn't come. Try using:
`git --no-pager`
It can be pass to the `$GIT` like `-C` option

Reply:
For me, github's shell trim some output and probably here is the problem.